components/bm1397/bm1397.c

```diff
@@ -132,7 +132,7 @@ void BM1397_send_hash_frequency(float frequency)
     float deffreq = 200.0;
 
     float fa, fb, fc1, fc2, newf;
-    float f1, basef, famax = 0xf0, famin = 0x10;
+    float f1, basef, famax = 0x104, famin = 0x10;
```

Comment:
this is not enough, for fa > 0xff, on line 188 below, the value will be truncated to 8bits and stored in freqbuf[3], so you should also adapt the line 188 to not loose the 9th bit (msb)

Reply:
updates as discussed